ndonnx/_core/_shapeimpl.py

```diff
@@ -66,6 +66,25 @@ def expand_dims(self, x, axis):
             )
         )
 
+    def repeat(self, x, repeats, axis):
+        if axis is None:
+            x = ndx.reshape(x, [-1])
+            axis = 0
+
+        x_shape = ndx.additional.shape(x)
+
+        if isinstance(repeats, int):
+            # TODO: this case can be optimized by broadcasting and reshaping
+            repeats = ndx.asarray(repeats)
+
+        repeats = ndx.broadcast_to(repeats, ndx.reshape(x_shape[axis], [1]))
+        indices = ndx.searchsorted(
+            ndx.cumulative_sum(repeats).astype(ndx.uint64),
+            ndx.arange(ndx.sum(repeats)),
+            side="right",
+        )
+        return ndx.take(x, indices, axis=axis)
```

Comment:
Nice trick! Not sure if it's what you already allude to in your TODO but I believe the cumulative sum search is only required for the case where you have different repeat values. If `repeat` is a python scalar, we can skip the very expensive `searchsorted`. We can revisit this in a future PR but it'll be useful to know how much performance we're giving up.